hypershift-operator/controllers/nodepool/capi.go

```diff
@@ -0,0 +1,1340 @@
+	var timeOut time.Duration
+
+	switch nodePool.Spec.Platform.Type {
+	case hyperv1.AgentPlatform, hyperv1.NonePlatform:
+		timeOut = 16 * time.Minute
+	default:
+		timeOut = 8 * time.Minute
+	}
+
+	maxUnhealthyOverride := nodePool.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
+	if maxUnhealthyOverride == "" {
+		maxUnhealthyOverride = hc.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
+	}
+	if maxUnhealthyOverride != "" {
+		maxUnhealthyValue := intstr.Parse(maxUnhealthyOverride)
+		// validate that this is a valid value by getting a scaled value
+		if _, err := intstr.GetScaledValueFromIntOrPercent(&maxUnhealthyValue, 100, true); err != nil {
+			log.Error(err, "Cannot parse max unhealthy override duration", "value", maxUnhealthyOverride)
+		} else {
+			maxUnhealthy = maxUnhealthyValue
+		}
+	}
+
+	timeOutOverride := nodePool.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
+	if timeOutOverride == "" {
+		timeOutOverride = hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
+	}
+	if timeOutOverride != "" {
+		timeOutOverrideTime, err := time.ParseDuration(timeOutOverride)
+		if err != nil {
+			log.Error(err, "Cannot parse timeout override duration", "value", timeOutOverride)
+		} else {
+			timeOut = timeOutOverrideTime
+		}
+	}
+
+	resourcesName := generateName(capiClusterName, nodePool.Spec.ClusterName, nodePool.GetName())
+	mhc.Spec = capiv1.MachineHealthCheckSpec{
+		ClusterName: capiClusterName,
+		Selector: metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				resourcesName: resourcesName,
+			},
+		},
+		UnhealthyConditions: []capiv1.UnhealthyCondition{
+			{
+				Type:   corev1.NodeReady,
+				Status: corev1.ConditionFalse,
+				Timeout: metav1.Duration{
+					Duration: timeOut,
+				},
+			},
+			{
+				Type:   corev1.NodeReady,
+				Status: corev1.ConditionUnknown,
+				Timeout: metav1.Duration{
+					Duration: timeOut,
+				},
+			},
+		},
+		MaxUnhealthy: &maxUnhealthy,
+		NodeStartupTimeout: &metav1.Duration{
+			Duration: 20 * time.Minute,
+		},
+	}
+	return nil
+}
+
+// setMachineDeploymentReplicas sets wanted replicas:
+// If autoscaling is enabled we reconcile min/max annotations and leave replicas untouched.
+func setMachineDeploymentReplicas(nodePool *hyperv1.NodePool, machineDeployment *capiv1.MachineDeployment) {
+	if machineDeployment.Annotations == nil {
+		machineDeployment.Annotations = make(map[string]string)
+	}
+
+	if isAutoscalingEnabled(nodePool) {
+		// The MachineDeployment replicas field should default to a value inside the (min size, max size) range based on the autoscaler annotations
+		// so the autoscaler can take control of the replicas field.
+		//
+		// 1. if it’s a new MachineDeployment, or the replicas field of the old MachineDeployment is < min size, use min size
+		// 2. if the replicas field of the old MachineDeployment is > max size, use max size
+		mdReplicas := k8sutilspointer.Int32Deref(machineDeployment.Spec.Replicas, 0)
```

Comment:
Int32Deref is deprecated.

Reply:
Thanks! fwiw this is just moving existing code around, intentionally modifying as less as possible in this PR and increasing coverage to an acceptable bar to enable further changes. I dropped all references to the deprecated package